[Yinze] iP

YINZE66 opened the pull request:
### GenjiPro
> “Your mind is for having ideas, not holding them.” – David Allen

GenjiPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/476#).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer,  you can use it to practice Java too. Here's the `main`  method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
add Level-0

Commit:
add hiii

Commit:
add goodbye

Commit:
Change file name to Genji.java

Commit:
Rename Duke.java

Commit:
Extend the line

Commit:
Add method addList

Commit:
Add method showList

Commit:
Add class Task.java

Commit:
add mark and unmark ability

Commit:
Add class ToDo.java which extends Task.java

Commit:
Add class Deadline.java which extends Task.java

Commit:
Add class Event.java which extends Task.java

Commit:
Fix some details

Commit:
Add todo and deadline to the chatbot

Commit:
Add event to the chatbot and fix some details

Commit:
Delete useless line in ToDo.java

Commit:
Update A-TextUiTesting

Commit:
Update different kind of exceptions and fix some details

Commit:
Add a delete method and fix some details

Commit:
Add saveList method and loadList method

Commit:
Fix details and handle cases

Commit:
Add this to create merge commit

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Fix details and handle cases
  Add saveList method and loadList method

Commit:
Try some test

Commit:
Add date time to tasks

Commit:
Handle some exceptions

Commit:
Add checkDate method to print deadlines/events occurring on a specific date.

Commit:
Add this to create merge commit

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add checkDate method to print deadlines/events occurring on a specific date.
  Handle some exceptions
  Add date time to tasks

Commit:
Add Ui class to deal with interactions
Add Storage class to load tasks
Add parser class to parse commands
Add TaskList lass to contain tasks

Commit:
Add command classes and exception class
Fix many details so that it is more OOP

Commit:
Add a line to input.txt
Do this to create merge commit

Commit:
Merge branch 'branch-MoreOOP'

* branch-MoreOOP:
  Add command classes and exception class Fix many details so that it is more OOP
  Add Ui class to deal with interactions Add Storage class to load tasks Add parser class to parse commands Add TaskList lass to contain tasks

Commit:
Add classes to different packages
Create genji, genji.command, and genji.task

df-Nic commented on src/main/java/genji/Genji.java:
```diff
@@ -0,0 +1,39 @@
+import genji.task.TaskList;
+
+public class Genji {
+    private Ui ui;
```
Possibly add a description for all class attributes to inform others what this variable is about.

**For example:**
```java
public class Genji {
    /** Handle displaying of UI elements to user */
    private Ui ui;

    // Rest of the class
}
```

df-Nic commented on src/main/java/genji/Genji.java:
```diff
@@ -0,0 +1,39 @@
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+    private static String FILE_PATH = "./data/Genji.txt";
```
Good that for static variables you used the snake case in caps as the variable name since it follows the coding standards.

df-Nic commented on src/main/java/genji/GenjiException.java:
```diff
@@ -0,0 +1,11 @@
+package genji;
+
+public class GenjiException extends Exception{
```
Add a whitespace after the `Exception` to follow the coding standard.

**For example:**
```java
public class GenjiException extends Exception {
    // Rest of the class
}
```

df-Nic commented on src/main/java/genji/Parser.java:
```diff
@@ -0,0 +1,120 @@
+package genji;
+
+import genji.command.Command;
```
It is good that the imports follow a specific standard determined by you for this and the other files as well 👍

df-Nic commented on src/main/java/genji/Parser.java:
```diff
@@ -0,0 +1,120 @@
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    public static Command parse(String command) throws GenjiException{
```
Similar as before add a whitespace between the `GenjiException`  and the  { to follow the coding standard.

df-Nic commented on src/main/java/genji/Genji.java:
```diff
@@ -0,0 +1,39 @@
+import genji.command.Command;
+import genji.task.TaskList;
+
+public class Genji {
```
Remember to add a class description header for JavaDocs to describe briefly what the class is.

df-Nic commented on src/main/java/genji/Storage.java:
```diff
@@ -0,0 +1,81 @@
+       this.filePath = filePath;
+   }
+
+    public TaskList loadList() {
```
For **public functions** for classes remember to add JavaDocs.

**For example:**
```java
/**
 * <Function Description>.
 *
 * @return <What does it return>.
 */
public TaskList loadList()
    // Function 
}
```

df-Nic commented on src/main/java/genji/Storage.java:
```diff
@@ -0,0 +1,81 @@
+                        list.get(list.size()- 1).mark();
+                    }
+                } else if (input.startsWith("D")) {
+                    String name = input.substring(8, input.lastIndexOf(" |"));
```
Possibly a small suggestion to add lines between code to declutter it, making it more readable.

df-Nic commented on src/main/java/genji/Ui.java:
```diff
@@ -0,0 +1,59 @@
+        this.scanner = new Scanner(System.in);
+    }
+
+    public void welcome() {
```
Function name should start with a verb as mention in the coding standards.

**For example:**
```java
public void printWelcome() {
    showLine();
    System.out.println("Hello! I'm Genji\nWhat can I do for you?  O.o?");
    showLine();
}
```

df-Nic commented on src/main/java/genji/command/AddCommand.java:
```diff
@@ -0,0 +1,25 @@
+import genji.task.TaskList;
+import genji.Ui;
+import genji.Storage;
+public class AddCommand extends Command{
```
Similar to before add a whitespace between `Command` and { .

df-Nic commented on src/main/java/genji/GenjiException.java:
```diff
@@ -0,0 +1,11 @@
+    public GenjiException(String s) {
+        super(s);
+    }
+    /*@Override
```
Possibly remove this code chunk since it is not in use?

df-Nic commented on src/main/java/genji/Parser.java:
```diff
@@ -0,0 +1,120 @@
+
+public class Parser {
+    public static Command parse(String command) throws GenjiException{
+        if (command.equals("bye")) {
```
It is good that you follow the indentation format for `if`, `else` statements as well as `try`, `catch` statements 💯

df-Nic reviewed (commented):
Overall you did a good job in following the coding standards as mention in the lecture materials 🥇 

Maybe take a look at some of the comments I made which I feel can improve the code quality

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Set up gradle to build an run Genji
Add two .xml file

Commit:
Add some JUnit tests
Test normal situation
Test situation with invalid input

Commit:
Add archiveFileName genji.jar
Build a JAR file using Gradle

Commit:
Add javaDoc to every classes
Fix some details

Commit:
Delete a useless line in GenjiException
Fix code style to reach standard

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javaDoc to every classes Fix some details

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Delete a useless line in GenjiException Fix code style to reach standard

Commit:
Add Level-9 features
Find a task using contain method
Fix some details about IndexOutOfBoundsException

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Level-9 features Find a task using contain method Fix some details about IndexOutOfBoundsException

# Conflicts:
#	src/main/java/genji/Ui.java
#	src/main/java/genji/command/DeleteCommand.java
#	src/main/java/genji/task/Task.java

Commit:
Add JavaDoc to new codes
Fix a missed merge conflict in last commit

Commit:
Fix some coding style related problems
Add space after the extending a class
Change a function name to start with a verb

Shruthiiii03 commented on src/main/java/genji/Genji.java:
```diff
@@ -0,0 +1,55 @@
+        this.taskList = storage.loadList();
+    }
+
+    /**
```
Adding on to the JavaDocs section, maybe you can consider adding the parameters and the result to the function as well. It might make the description of the code more comprehensive and easily understandable by other developers. For example, just by doing /** and entering, the template for the method javadocs is automatically shown in code editors like IntelliJ.

Shruthiiii03 commented on src/main/java/genji/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    public static Command parse(String command) throws GenjiException {
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
```
Good use of Egyptian brackets, proper indentations and spacings! The code style has been well-followed 👍

Shruthiiii03 commented on src/main/java/genji/command/DateCommand.java:
```diff
@@ -0,0 +1,40 @@
+package genji.command;
```
I like that you split your commands into separate smaller classes. It makes the code much more readable and follows the rules of encapsulation, making your code cleaner as well.

Shruthiiii03 reviewed (commented):
Overall, I found your code pretty easy to read and well-encapsulated. Regarding the javaDocs, maybe making it a little more comprehensive, will enable readers to understand the code and its function much more easily. Apart from that, I believe the code is well done!

Commit:
Add JavaFX related classes
Add DaGenji.png and DaUser.png to images

Commit:
Add getResponse method to print related message
Add JavaDoc to new classes and methods
Fix some nits

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add getResponse method to print related message Add JavaDoc to new classes and methods Fix some nits
  Add JavaFX related classes Add DaGenji.png and DaUser.png to images

Commit:
Add assert to various points in the code

Need to document assumptions that should hold for some points

Using assertions to define assumptions

Assertion is a convenient way to check program state

Commit:
Modify method in Parser.java and Storage.java

Meet java code quality

Splitting conditions into methods thus avoiding too long
method and deep nesting

Improve code quality makes the program better

YINZE66 committed:
Merge pull request #1 from YINZE66/branch-A-Assertions

Add assert to various points in the code

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assert to various points in the code

YINZE66 committed:
Merge pull request #2 from YINZE66/branch-A-CodeQuality

Modify method in Parser.java and Storage.java

Commit:
Add C-Help feature to the program

Help users be familiar with commands

Printing out the commands and related formats

Explicitly help users

Commit:
Fix some nits and code quality related stuff

Commit:
Fix some documentation to increase code quality

Commit:
Add screenshot of Ui Ui.png to docs

Commit:
Add Genji user guide to README.md

Commit:
Add an omitted feature to the user guide in README.md

Commit:
Add more JUnit tests to ParserTest.java

Test more features of Genji whether they have bugs

Importing JUnit to test find and date command

JUnit test is convenient and can import directly

Commit:
Add the function of exiting GUI when user input bye

Commit:
Fix the exit feature with delay to show bye message

Commit:
Fix a nit in README.md related to Ui.png issue
